zed/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+SET(execName ZED_with_OpenCV.so)
+CMAKE_MINIMUM_REQUIRED(VERSION 2.4)
+PROJECT(${execName})
+
+option(LINK_SHARED_ZED "Link with the ZED SDK shared executable" ON)
+
+if (NOT LINK_SHARED_ZED AND MSVC)
+    message(FATAL_ERROR "LINK_SHARED_ZED OFF : ZED SDK static libraries not available on Windows")
+endif()
+
+if(COMMAND cmake_policy)
+	cmake_policy(SET CMP0003 OLD)
+	cmake_policy(SET CMP0015 OLD)
+endif(COMMAND cmake_policy)
+
+SET(EXECUTABLE_OUTPUT_PATH ".")
+
+find_package(ZED 3 REQUIRED)
+find_package(OpenCV REQUIRED)
+find_package(CUDA ${ZED_CUDA_VERSION} EXACT REQUIRED)
+
+IF(NOT WIN32)
+    add_definitions(-Wno-format-extra-args)
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+    SET(SPECIAL_OS_LIBS "pthread" "X11")
+ENDIF()
+
+include_directories(${CUDA_INCLUDE_DIRS})
+include_directories(${ZED_INCLUDE_DIRS})
+include_directories(${OpenCV_INCLUDE_DIRS})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+link_directories(${ZED_LIBRARY_DIR})
+link_directories(${OpenCV_LIBRARY_DIRS})
+link_directories(${CUDA_LIBRARY_DIRS})
+
+FILE(GLOB_RECURSE SRC_FILES src/main.cpp)
+FILE(GLOB_RECURSE HDR_FILES include/*.hpp)
+
+ADD_EXECUTABLE(${execName} ${HDR_FILES} ${SRC_FILES})
+add_definitions(-std=c++17 -g -O3)
```

Comment:
Shouldn't `DCMAKE_BUILD_TYPE` take care of setting -g and -O3?

Reply:
I removed this.